src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -786,17 +803,112 @@ def load_module(name, value):
         # import it here to avoid circular import
         from diffusers import pipelines
 
-        # 6. Load each module in the pipeline
+        ###### 6. device map delegation ######
+        final_device_map = None
+        if device_map is not None and device_map.startswith("balanced"):
```

Comment:
Make sure also that if the user pass a `device_map` that is not in ["balanced"], we return an error. 
```suggestion
        if isinstance(device_map,str): 
```